### PR TITLE
TP-1027: In-memory GigaSpaces Manager, LUS & Zookeeper

### PR DESCRIPTION
**Pull request by pativa**
* This contains *BREAKING* changes due to internal GigaSpaces workings. Due to this, this PR suggests that we change the next release version to `2.1.0`.
* The intention is to use an in-memory version of GigaSpaces Manager that starts a LUS and Zookeeper, also in-memory.
* Zookeeper is started by `curator-test`, which is the same dependency that is used by GigaSpaces for tests.
* This library now has a dependency to Zookeeper & Curator. However, usage of Zookeeper in tests can be disabled by a system property. We could default to not use Zookeeper, but I thought it best to enable it by default.
* Additionally, this also changes this library from using `multicast` to `unicast`. This is to make the tests more production-like for the Avanza environment. This is a breaking change and any usage of `JVMGlobalLus` has to be migrated. Support for using `multicast` has been removed.

---
This PR is targeted at https://github.com/AvanzaBank/gs-test/pull/37 to make it clear which changes are part of this PR.

### CODE REVIEW

File: src/main/resources/zoo.cfg.template [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# clientPort:
+#     the port at which the clients will connect
+clientPort=##CLIENT_PORT##
```
Comment (pativa) [inline src/main/resources/zoo.cfg.template:3]: The `zoo.cfg` generated file is only used by GigaSpaces and its not used for configuring the Zookeeper server instance itself. I have made this file minimal, containing only the `clientPort` used for finding the Zookeeper instance by GS.

File: src/test/java/com/avanza/gs/test/ZookeeperLeaderSelectorTest.java [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+				.numberOfBackups(1)
+				.configure()) {
+			fruitPu.start();
+
+			ApplicationContext primaryContext = fruitPu.getPrimaryInstanceApplicationContext(0);
+			ApplicationContext backupContext = fruitPu.getBackupInstanceApplicationContext(0, 1);
+
+			getClusteredGigaSpace(primaryContext).write(new FruitPojo("apple"));
+
+			// Verify that primary & backup are configured as expected first
```
Comment (Bjornelmers) [inline src/test/java/com/avanza/gs/test/ZookeeperLeaderSelectorTest.java]: I really like these detailed test comments! 👍

**Review by Bjornelmers — APPROVED** [review]
Very good @pativa! 
One minor suggestion however: I think Zookeeper should be in came case (ZooKeeper). Apache seems to spell it like that at least (GigaSpaces however are inconsistent in their spelling).

**Comment by pativa** [comment]
> Very good @pativa! One minor suggestion however: I think Zookeeper should be in came case (ZooKeeper). Apache seems to spell it like that at least (GigaSpaces however are inconsistent in their spelling).

Good suggestion @Bjornelmers! I've updated this PR to use 'ZooKeeper' instead of 'Zookeeper' in class names, variable names, comments and documentation 👍

**Review by askoog — APPROVED** [review]

**Review by kbs4j — APPROVED** [review]
Looks good @pativa ! 👍

**Pull request merged**